### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by michielbdejong**
r? @samgiles

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+RUN curl -fsOSL $RUST_DOWNLOAD_URL \
+    && curl -s $RUST_DOWNLOAD_URL.sha256 | sha256sum -c - \
+    && tar -C /rust -xzf $RUST_ARCHIVE --strip-components=1 \
+    && rm $RUST_ARCHIVE \
+    && ./install.sh
+
+COPY . /rust
+RUN cd /rust && cargo build
+CMD service redis-server start && RUST_LOG=info ./target/debug/registration_server -h 0.0.0.0 -p 4443 --cert-directory /certdir
```
Comment (samgiles) [inline Dockerfile]: We shouldn't really run redis and the server in the same docker container.  Instead we should link to the redis server and use environment variables to configure.  Will make a follow up issue so we're not blocking on this.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+WORKDIR /rust
+
+RUN curl -fsOSL $RUST_DOWNLOAD_URL \
+    && curl -s $RUST_DOWNLOAD_URL.sha256 | sha256sum -c - \
+    && tar -C /rust -xzf $RUST_ARCHIVE --strip-components=1 \
+    && rm $RUST_ARCHIVE \
+    && ./install.sh
+
+COPY . /rust
+RUN cd /rust && cargo build
```
Comment (samgiles) [inline Dockerfile]: It's a shame we can't cross compile and have the tiniest docker container. alpine + binary.  Will make a follow up issue.

File: Dockerfile (hunk 1 above)
Comment (samgiles) [inline Dockerfile]: We should also build release, not debug.

File: Dockerfile (hunk 2 above)
Comment (samgiles) [inline Dockerfile]: #27

File: Dockerfile (hunk 1 above)
Comment (samgiles) [inline Dockerfile]: First comment: #26

**Comment by michielbdejong** [comment]
Changed it to a release build, will test it on the server before merging.

**Comment by michielbdejong** [comment]
I left it building in a screen on the server, will update the live container to the new image when the build is done, check everything still works, then merge this PR.

**Pull request merged**